rust/src/smb/smb1_records.rs

```diff
@@ -106,15 +107,23 @@ pub fn parse_smb1_write_andx_request_record(i : &[u8], andx_offset: usize) -> IR
     let (i, _remaining) = le_u16(i)?;
     let (i, data_len_high) = le_u16(i)?;
     let (i, data_len_low) = le_u16(i)?;
+    // data_offset: from the start of SMB Header to the start of data
     let (i, data_offset) = le_u16(i)?;
     let (i, high_offset) = cond(wct == 14, le_u32)(i)?;
-    let (i, bcc) = le_u16(i)?;
+    let (i, _bcc) = le_u16(i)?;
     //spec [MS-CIFS].pdf says always take one byte padding
-    let (i, _padding) = cond(bcc > data_len_low, |b| take(bcc - data_len_low)(b))(i)?; // TODO figure out how this works with data_len_high
-    let (i, _padding_evasion) = cond(data_offset > ax+4+2*(wct as u16), |b| take(data_offset - (ax+4+2*(wct as u16)))(b))(i)?;
+    let (i, _padding) = take(1_usize)(i)?;
+    // [MS-CIFS].pdf 2.2.4.43.1
+    // without high_offset, &SMB_Data.Bytes.Data - &SMB_Parameters = 28
+    // with high_offset, &SMB_Data.Bytes.Data - &SMB_Parameters = 32
+    let mut data_pad_offset = data_offset.saturating_sub(ax+28);
+    if high_offset.is_some() {
+        data_pad_offset = data_pad_offset.saturating_sub(4);
```

Comment:
I wonder if we can be more direct by using `wct` as was done before (`ax+4+2*(wct as u16)`)